src/CadViewer.tsx

```diff
@@ -18,10 +20,31 @@ export const CadViewer = (props: any) => {
     setMenuVisible,
   } = useContextMenu({ containerRef })
 
+  // GLTF export functionality
+  const [sceneRef, saveGltfAs] = useSaveGltfAs()
+
+  function useCombinedRefs<T>(...refs: (React.Ref<T> | undefined)[]) {
```

Comment:
This function should not be inside a render function, and it shouldn't be defined in such a central component